indico/modules/events/timetable/templates/pdf/_program.html

```diff
@@ -0,0 +1,340 @@
+{% endmacro %}
+
+{% macro render_session_block(entry, config) %}
+    {% set session_block = entry.object %}
+    {% set sess = session_block.session %}
+    {% set conveners = session_block.person_links %}
+    {% set contributions = session_block.contributions %}
+    {% set show_session_block_content = entry.children or (config.show_session_description and sess.description) %}
+
+    <section
+        id="entry-{{ entry.id }}"
+        class="session-block-title inherited-bg-color inherited-txt-color {{ 'only-round-top' if show_session_block_content else ''}}"
+    >
+        {{ render_entry_header(entry) }}
+    </section>
+    {% if show_session_block_content %}
+        <div class="session-block-content">
+            {% if config.show_session_description and sess.description %}
+                {{ render_description(sess.description) }}
+            {% endif %}
+            {% if entry.children %}
+                {{ render_block_child_entries(sess, entry.children, config) }}
+            {% endif %}
+        </div>
+    {% endif %}
+{% endmacro %}
+
+{% macro render_block_child_entries(sess, children, config) %}
+    <div class="session-block-content-box inherited-border-color">
+        <ul class="contrib-timeline">
+            {% for child in children %}
+                <li class="inherited-border-color">
+                    <div class="contrib-heading">
+                        {% if not sess.is_poster %}
+                            <span class="timebox inherited-bg-color inherited-txt-color">
+                                {{ format_interval(child.start_dt.astimezone(child.event.tzinfo), child.end_dt.astimezone(child.event.tzinfo), format='HHmm') }}
+                                {% if config.show_length_contribs %}
+                                    ({{ (child.end_dt - child.start_dt)|format_human_timedelta(narrow=True) }})
+                                {% endif %}
+                            </span>
+                        {% endif %}
+                        <h4>{{ child.object.title }}</h4>
+                    </div>
+                    {% if child.type.name == 'CONTRIBUTION' %}
+                        <div class="contrib-timeline-content" >
+                            {{ render_child_contrib(child.object, config) }}
+                        </div>
+                    {% endif %}
+                </li>
+            {% endfor %}
+        </ul>
+    </div>
+{% endmacro %}
+
+{% macro render_child_contrib(contrib, config) %}
+    {% set speakers = contrib.speakers %}
+
+    {% if speakers %}
+        <p>
+            <span class="bold">{{ ngettext('Speaker', 'Speakers', speakers|length) }}</span><br />
+            {{ render_names(speakers, show_title=config.show_title, show_affiliation=config.show_affiliation) }}
+        </p>
+    {% endif %}
+
+    {% if not contrib.inherit_location and contrib.has_location_info %}
+        <p>
+            <span class="bold">{% trans %}Location{% endtrans %}</span><br />
```

Comment:
Unless weasyprint requires XHTML-style self-closing tags, let's avoid them since we don't use them elsewhere.

```suggestion
            <span class="bold">{% trans %}Location{% endtrans %}</span><br>
```